### PR TITLE
Use of a substations hierarchy for ordering generators in generation dispatch

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/components/utils/rhf-inputs/chip-items-input.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import MidFormError from './error-inputs/mid-form-error';
+import { RawReadOnlyInput } from './read-only/raw-read-only-input';
+import { FieldLabel } from '../inputs/hooks-helpers';
+import { isFieldRequired } from '../utils';
+
+const ChipItemsInput = ({ label, name, hideErrorMessage }) => {
+    const [textEntered, setTextEntered] = useState('');
+    const { snackError } = useSnackMessage();
+
+    const classes = useStyles();
```
Comment (dbraquart) [inline src/components/utils/rhf-inputs/chip-items-input.js:26]: can you use sx instead ?

File: src/components/utils/rhf-inputs/chip-items-input.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                    InputProps={{
+                        disableUnderline: true,
+                        style: {
+                            height: '30px',
+                            left: '5px',
+                        },
+                    }}
+                    value={textEntered}
+                    onKeyDown={keyPress}
+                    onChange={handleChange}
```
Comment (dbraquart) [inline src/components/utils/rhf-inputs/chip-items-input.js:124]: maybe onBlur here to add an item without Enter ?

File: src/components/dialogs/dialogUtils.js [hunk 3]
```diff
@@ -71,6 +71,28 @@ export const useStyles = makeStyles((theme) => ({
     addDirectoryElements: {
         marginTop: -5,
     },
+    formChipItems1: {
```
Comment (dbraquart) [inline src/components/dialogs/dialogUtils.js:74]: maybe more explicit names rather than xxx1 et xxx2 ?

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: just a few remarks
tests: done. some errors and questions:

- when I try to open a former 'Generator Dispatch' (where substationsGeneratorsOrdering is null in Get request), I got a crash in study-app. Browser Logs:
	_Uncaught TypeError: arrayToWatch is undefined
		allRowSelected dnd-table.js:57
	Uncaught TypeError: currentRows is undefined
		DndTableBottomRightButtons dnd-table-bottom-right-buttons.js:36_
- Same if I create a new 'Generator Dispatch' without any 'Substations hierarchy'
- Adding a Substation name: when we do not strike enter after striking a Substation name, it is not taken into account.
	Maybe we should do something on event 'onBlur' to validate a Substation, with or without striking Enter.
- When I enter an unexisting Substation, this log is made N times: _INFO: Substation SYYY not found_
	- Maybe a single time ?
	- Maybe a Warning would be better ?

File: src/components/utils/rhf-inputs/chip-items-input.js (hunk 2 above)
Comment (FranckLecuyer) [reply src/components/utils/rhf-inputs/chip-items-input.js:124]: Done

**Comment by FranckLecuyer** [comment]
> code: just a few remarks tests: done. some errors and questions:
> 
>     * when I try to open a former 'Generator Dispatch' (where substationsGeneratorsOrdering is null in Get request), I got a crash in study-app. Browser Logs:
>       _Uncaught TypeError: arrayToWatch is undefined
>       allRowSelected dnd-table.js:57
>       Uncaught TypeError: currentRows is undefined
>       DndTableBottomRightButtons dnd-table-bottom-right-buttons.js:36_
> 
>     * Same if I create a new 'Generator Dispatch' without any 'Substations hierarchy'
> 
>     * Adding a Substation name: when we do not strike enter after striking a Substation name, it is not taken into account.
>       Maybe we should do something on event 'onBlur' to validate a Substation, with or without striking Enter.
> 
>     * When I enter an unexisting Substation, this log is made N times: _INFO: Substation SYYY not found_
>       
>       * Maybe a single time ?
>       * Maybe a Warning would be better ?

File: src/components/utils/rhf-inputs/chip-items-input.js (hunk 1 above)
Comment (FranckLecuyer) [reply src/components/utils/rhf-inputs/chip-items-input.js]: Done

File: src/components/dialogs/dialogUtils.js (hunk 3 above)
Comment (FranckLecuyer) [reply src/components/dialogs/dialogUtils.js]: Removed (use of sx instead : see above)

**Comment by FranckLecuyer** [comment]
> 

All corrections and modifications made (some in back side in modification-server)

File: src/components/utils/rhf-inputs/chip-items-input.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                //     [classes.formChipItemsError]: error?.message,
+                // })}
+
+                sx={{
+                    display: 'flex',
+                    gap: '8px',
+                    flexWrap: 'wrap',
+                    flexDirection: 'row',
+                    border: '2px solid lightgray',
+                    padding: 1,
```
Comment (ayolab) [inline src/components/utils/rhf-inputs/chip-items-input.js:86]: maybe you can remove the padding here ?

File: src/components/utils/rhf-inputs/chip-items-input.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    };
+
+    const handleChange = (e) => {
+        setTextEntered(e.target.value);
+    };
+
+    return (
+        <>
+            <FormControl
+                // className={clsx(classes.formChipItems1, {
```
Comment (ayolab) [inline src/components/utils/rhf-inputs/chip-items-input.js:76]: remove this commented code and maybe you can regroup styles in a const 
for example : 
    const styles = {
        chipContainer: {
                    display: 'flex',...
         },
        chipItem: {
        ....
        }

and then use them in sx : sx={styles.chipItem}

**Review by ayolab — APPROVED** [review]

File: src/components/utils/rhf-inputs/chip-items-input.js (hunk 4 above)
Comment (FranckLecuyer) [reply src/components/utils/rhf-inputs/chip-items-input.js]: Removed

File: src/components/utils/rhf-inputs/chip-items-input.js (hunk 5 above)
Comment (FranckLecuyer) [reply src/components/utils/rhf-inputs/chip-items-input.js]: Done

**Review by dbraquart — APPROVED** [review]
code: ok
tests: ok

**Pull request merged**